pilot/pkg/model/push_context.go

```diff
@@ -695,6 +695,21 @@ func (ps *PushContext) ServiceForHostname(proxy *Proxy, hostname host.Name) *Ser
 	return nil
 }
 
+// ServicesForHostname returns a list of services that fall under the hostname provided. This hostname
+// can be a wildcard.
+func (ps *PushContext) ServicesForHostname(proxy *Proxy, hostname host.Name) []*Service {
+	if proxy == nil || proxy.SidecarScope == nil {
+		return nil
+	}
+	services := make([]*Service, 0)
+	for _, svc := range proxy.SidecarScope.services {
```

Comment:
I would rather do this only when the given `hostname` is wildcarded

Reply:
I don't understand, what's the alternative? a lookup?

Reply:
I mean the perf can be improved